tidy3d/components/medium.py

```diff
@@ -1088,6 +1146,16 @@ def components(self) -> Dict[str, Medium]:
         """Dictionary of diagonal medium components."""
         return dict(xx=self.xx, yy=self.yy, zz=self.zz)
 
+    @cached_property
+    def n_cfl(self):
+        """This property computes the index of refraction related to CFL condition, so that
+        the FDTD with this medium is stable when the time step size that doesn't take
+        material factor into account is multiplied by ``n_cfl``.
+
+        For this medium, it takes the minimal of ``n_clf`` in all components.
+        """
+        return min((mat_component.n_cfl for mat_component in self.components.values()))
```

Comment:
I think this is fine, but conceptually, do you know if this can be loosened? For example for mesh step, we take some sort of average between the mesh steps along each of the three directions.

Reply:
For uniform medium, it should be "courant/sqrt[sum_i 1/(n_cfl[i]*dl_min[i])**2]/C_0". But in my test, it diverges when the simulation domain is no longer uniform.